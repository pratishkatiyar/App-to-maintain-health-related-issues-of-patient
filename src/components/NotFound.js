import React from 'react'
import { Link } from 'react-router-dom'
import Navbar from './NavBar';

function NotFound() {
    return (
        <div data-testid='notFound-1'>
            <Navbar/>
            <h1>
                404 error<br/>Page not found.<br/>
            </h1>            
        </div>
    )
}

export default NotFound
