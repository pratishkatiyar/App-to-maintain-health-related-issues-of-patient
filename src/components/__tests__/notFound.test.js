import { render, screen, cleanup } from '@testing-library/react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import NotFound from '../NotFound'

afterEach(() => {
    cleanup();
})

test('should render NotFound component', () =>{
    render(<BrowserRouter><NotFound/></BrowserRouter>);
    const notFoundElement = screen.getByTestId('notFound-1');
    expect(notFoundElement).toBeInTheDocument();
    expect(notFoundElement).toHaveTextContent('404 error');
    expect(notFoundElement).toHaveTextContent('Page not found.');
})

test('matches snapshot', ()=> {
    const tree = renderer.create(<BrowserRouter><NotFound/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})