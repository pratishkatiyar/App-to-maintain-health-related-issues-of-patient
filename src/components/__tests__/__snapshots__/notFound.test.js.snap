// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="notFound-1"
>
  <div
    data-testid="navbar-1"
  >
    <nav>
      <div
        className="logo"
      >
        <p>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </p>
      </div>
      <div
        className="logo"
      >
        Patient Registration App
      </div>
      <div />
    </nav>
  </div>
  <h1>
    404 error
    <br />
    Page not found.
    <br />
  </h1>
</div>
`;
